pingcap/tidb
# correct  config.toml.example

### What problem does this PR solve? 
the configuration `check-mb4-value-in-utf8` will never take effect in `conf/config.toml.example`

### What is changed and how it works?
put `check-mb4-value-in-utf8` into right section of config file according to[ the syntax of the toml configuration file](https://github.com/toml-lang/toml#table)

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   * create a config file,  set `check-mb4-value-in-utf8 = false`
   * run tidb-server binary, check it takes effect

Related changes

 - Need to cherry-pick to the release branch
